### PR TITLE
Issue #953: Add condition-driven --batch --until mode to /auto

### LINKED ISSUE #953: auto: --batch に条件駆動 (--until) モードを追加
## Background

2026-07-05〜06 のセッションで、`/goal` (Claude Code 本体機能) を使い「project board Backlog の `retro/*` issue がゼロになるまで `/auto` で消化する」というオーケストレーションを実施した。

このセッションでは以下を **手動で** 行う必要があった:

1. 毎ラウンド、project board を再クエリして Backlog の `retro/*` issue を洗い出す (Python/jq をその都度手書き)
2. 各 issue を `run-issue.sh` → `get-issue-size.sh` → `run-auto-sub.sh` → `Skill(verify)` と個別にオーケストレーション (`/auto --batch` List mode の内部ロジックを issue ごとに手で再現)
3. `auto-checkpoint.sh write_batch`/`update_batch`/`delete_batch` を各ラウンドで手動呼び出し (`--batch` が内部で行っている処理と同一)
4. セッション完了後、L3 auto-retrospective (`docs/sessions/*/session.md`) が自動生成されなかったため、ユーザーの指摘を受けて事後に手動作成

つまり、既存の `/auto --batch` (Count mode / List mode) には「動的に作業対象を再発見しながら、意味的な条件が満たされるまで繰り返す」という第3のモードが存在せず、今回はそのギャップを issue 起票者 (Claude) が都度手作業で埋め合わせる形になった。同時進行中の別セッションが新規 `retro/*` issue を供給し続けたため、固定件数・固定リストの `--batch` では対応しきれない状況が実際に発生した。

### 補足 (2026-08-10 セッション横断レビューで確定): テーマ駆動での Backlog 消化がユースケースとして具体化

2026-08-10 のセッション横断レビューで、本 Issue の `--until` モードがそのまま応える運用要望が確認された。要望は「細かい指示は出さずに、特定テーマの Issue を `/auto --batch` で消化し、セッション中に起票された Issue も自動的に batch に追加・消化される」動作。

| 要望 | 本 Issue の該当箇所 |
|---|---|
| 特定テーマの Issue を選定 | `--until "label:..."` の label ベースクエリ |
| セッション中に起票された Issue の自動追加 | 「クエリ実行 → 処理 → **再度クエリ実行** → 0 件になるまで繰り返す」の再クエリ |
| 細かい指示を出さない | `--checkin-per-round` を opt-in とし、既定は確認なし |
| 無限ループの回避 | `--max-rounds` / wall-clock 上限を既定有効の安全弁として必須化 |

選定軸として `theme/*` label を追加する #1322 (`triage: テーマ label (theme/*) を導入し Backlog をテーマ単位で消化可能にする`) が別途起票済み。着地すると本 Issue のクエリは `/auto --batch --until "label:theme/observability" [--max-rounds 3]` の形になる。両 Issue は独立に着地可能 (本 Issue は既存の `retro/*` 等の label だけでも動作し、#1322 は `/triage` の分類として単独でも意味を持つ) であり、**blocked-by は設定しない**。

### 安全弁の必要性を裏付ける実測

2026-08-08 の session `23043-1786197225` で、「他セッションが処理速度を上回るペースで新規 issue を供給し続けるケース」が再観測された: 処理 7 件に対し、本 batch 中に起票された retro Issue は 8 件 (#1285 / #1287 / #1288 / #1289 / #1292 / #1293 / #1294 / #1300)。「batch 中に発生した retro Issue は remaining に自動追加」の慣例をそのまま適用すると消化と生成が釣り合い batch が終わらず、このときはユーザーが手動で区切りを判断して終了した (#1289 まで消化、残りは次セッション)。

一方、後続の session `97764-1786198856` では手動で 3 batch を連鎖させた結果、第 3 batch で新規起票ゼロとなり連鎖が 2 段で収束している。発散が常態ではないことの実測でもある。

出典: `docs/sessions/23043-1786197225-2026-08-09/session.md` § Findings、`docs/sessions/97764-1786198856-2026-08-09/session.md` § What worked

## Purpose

`/auto --batch` に条件駆動の第3モード `--until <query>` を追加し、今回手動で再実装した「動的発見 → 処理 → 再発見 → 空になるまで繰り返す」ループを `/auto` 本体の機能として提供する。既存の Count mode / List mode はそのまま維持し、廃止・統合はしない (それぞれ異なるユースケース — 「取り敢えず溜まってる分」「明示的なN件」— に応えているため)。

## Proposal (Outline)

```
/auto --batch --until "label:retro/* status:Backlog" [--max-rounds 3] [--checkin-per-round]
/auto --batch --until "label:theme/observability" [--max-rounds 3]
```

- **クエリ形式**: label + project board status のフィルタ (project board フィールドに対する構造化クエリ)。`status:` 句は省略可能で、label 単独指定 (例: `"label:theme/observability"`) でも動作する。今回の「Backlog の retro/*」のような条件を、LLM が毎回自然言語から再解釈するのではなく機械的に評価できる形にする
- **クエリの受け渡し**: `--until` の値は空白を含むため、必ずダブルクォートで囲んだ単一引数として渡す (`--until "label:retro/* status:Backlog"`)
- **ループ**: クエリ実行 → 該当 issue を List mode 相当のロジックで処理 → 再度クエリ実行 → 0件になるまで繰り返す
- **収束保証**: 同一 `--until` セッション内で処理済み (complete / fail) となった Issue はラウンドを跨いで保持し、次ラウンドのクエリ結果から除外する。label 単独クエリ (例: `label:retro/*`) では完了済み Issue も再ヒットするため、この除外がないとループが収束しない。blocked-by で skip した Issue は除外せず次ラウンドで再評価する (blocker が同一セッション中に解消しうるため)
- **安全弁 (必須)**: `--max-rounds` (既定値 3。他セッションが処理速度を上回るペースで新規 issue を供給し続けるケースが実際に観測されているため、無限ループ化を防ぐガードは opt-out ではなく既定動作とする) または wall-clock 上限のいずれか (両方でも可) を必ず備える。既定値 3 は 2026-08-08/09 の実測 (7件処理に対し新規8件供給という発散ケースと、3 batch 連鎖で2段収束した事例) を踏まえた実績ベースの初期値であり、運用実績次第で `/spec` 側の判断により調整可能とする
- **既存機構の再利用**: `auto-checkpoint.sh` の write/update/delete、Batch Completion Report、L3 auto-retrospective ( `docs/sessions/` への session.md 自動生成) は List mode のものをそのまま流用し、新規実装しない
- **ユーザーとの区切り確認**: `--checkin-per-round` で、ラウンド境界ごとに「継続するか」を尋ねるかどうかを選択可能にする。デフォルトは L3 autonomy の無停止実行方針に合わせて確認なしとし、明示的にオプトインさせる

### Pre-merge

- [x]  `skills/auto/SKILL.md` (または関連 module) に `--batch --until <query>` の処理ステップが実装されている
- [x]  安全弁 (max-rounds またはwall-clock上限) が既定で有効になっている
- [x]  安全弁 flag `--max-rounds` (既定値 3) が `skills/auto/SKILL.md` に定義されている
- [x]  既存の checkpoint・Batch Completion Report・L3 auto-retrospective 機構が新規実装なしで再利用されている
- [x]  クエリ解決スクリプト `scripts/resolve-batch-query.sh` の bats テストが PASS する
- [x]  `--until` / `--max-rounds` / `--checkin-per-round` の使用方法が `docs/workflow.md` に文書化されている

### Post-merge

- [ ] 次回 `--batch --until` を実運用で使用した際、動的に増減する対象を正しく処理し安全弁が意図通り機能することを観察 

## Related

- 発生元: 2026-07-05〜06 の `/goal` セッション (`docs/sessions/58088-1783222753-2026-07-06/session.md`)
- 対象 file: `skills/auto/SKILL.md`, `scripts/resolve-batch-query.sh` (新規)、`scripts/auto-checkpoint.sh` (再利用のみ、変更なし想定)
- Related to #1322 (`theme/*` label 導入。独立着地可能、blocked-by なし)
- Related to #1214 (next-cycle seed 撤去)。`--until` は seed file 方式を再導入せず、in-session 状態と既存 checkpoint のみで完結させる
- スコープ外: `/goal` 本体 (Claude Code CLI 機能) の Stop hook UX 改善は wholework リポジトリの対象外のため別途 Claude Code へフィードバック予定

## Autonomous Auto-Resolve Log

- **`--max-rounds` の既定値を 3 に設定** — reason: 2026-08-10 コメントで示された実測 (7件処理に対し新規8件供給という発散ケース、および3 batch連鎖で2段収束した事例) から「3〜5 程度から始めて実績で調整」という提案があり、least-risk (最も保守的で暴走を防ぎやすい) な下限値 3 を採用した。`/spec` の判断で調整可能な初期値として明記。
  - Other candidates: 5 (提案レンジの上限。処理量は増えるがラウンド当たりの安全マージンは小さくなる)
- **クエリ形式に label 単独指定 (status 句省略) を許容** — reason: #1322 のコメント例 `"label:theme/observability"` は status 句を含まないため、既存の "label + status" 前提の記述のままだと #1322 のユースケースと矛盾する。既存コードパターン (構造化クエリの各句は独立フィルタ) から、status 句は optional とするのが自然な拡張であり、Issue 記述をそれに合わせて明確化した。
  - Other candidates: status 句を必須のままにし、label 単独ケースは `status:*` 相当のワイルドカードで表現する (verbose になるため不採用)
- **`--checkin-per-round` の名称を確定 (「仮称」表記を削除)** — reason: 2026-08-10 コメントでの用語使用時にも同名がそのまま使われており、他に代替案が提示されていないため、確定名として扱う。Acceptance Criteria のテキストには影響しない。

### PR DESCRIPTION
**Pull request by saito**
## Summary

`/auto --batch` に条件駆動の第3モード `--batch --until <query>` を追加します。既存の Count mode (`--batch N`) / List mode (`--batch N1 N2 ...`) はそのまま維持し、「クエリ実行 → List mode 相当の処理 → 再クエリ → 0 件になるまで繰り返す」ループを新設します。

- `scripts/resolve-batch-query.sh` (新規): `label:`/`status:` クエリを決定的に解決するスクリプト。bash 3.2 互換。Issue 単位の Status GraphQL 解決失敗は fail-closed で除外
- `skills/auto/SKILL.md`: `### Until mode (--batch --until <query>)` を新設。ラウンドループは `auto-checkpoint.sh` の `write_batch`/`delete_batch` と `### List mode` の処理をそのまま再利用し、新規実装は行わない
- 安全弁 `--max-rounds` (既定 3、opt-out 不可) を必須で実装
- `docs/workflow.md` / `docs/ja/workflow.md` / `docs/guide/workflow.md` / `docs/ja/guide/workflow.md` / `docs/structure.md` / `docs/ja/structure.md` を更新

## Verification (pre-merge)

- rubric: `skills/auto/SKILL.md` に条件駆動ループが実装されている
- rubric: 既定で有効な安全弁 (`--max-rounds`) が実装されている
- `skills/auto/SKILL.md` に `--max-rounds` が定義されている
- rubric: `auto-checkpoint.sh` / L3 auto-retrospective を List mode から再利用し重複実装がない
- `bats tests/resolve-batch-query.bats` が PASS する (11/11)
- `docs/workflow.md` に `--until` の使用方法が文書化されている

## Verification (post-merge)

- 次回 `--batch --until` を実運用で使用した際、動的に増減する対象を正しく処理し安全弁が意図通り機能することを観察 (opportunistic)

## Notes

- `tests/post_merge_check.bats` の並列実行時フレーク (`bats --jobs` 実行時のみ再現、単体実行では PASS) を確認しましたが、既存 Issue #1308 で追跡済みのため新規起票はスキップしました。本 PR の変更 (`scripts/resolve-batch-query.sh` / `skills/auto/SKILL.md` / docs) とは無関係です。

closes #953

### CODE REVIEW

File: tests/resolve-batch-query.bats [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    export MOCK_ISSUE_LIST_JSON="$(three_issues_json)"
+    export MOCK_STATUS_BY_NUM="10:FAIL
+12:Backlog"
+    run --separate-stderr bash "$SCRIPT" --query "label:retro/* status:Backlog"
+    [ "$status" -eq 0 ]
+    [ "$output" = "12" ]
+    [[ "$stderr" == *"excluding"* ]]
+}
+
+@test "query with whitespace: accepted as a single quoted argument" {
```
Comment (saito) [inline tests/resolve-batch-query.bats:158]: [Spec deviation] The Spec's `## Uncertainty` section explicitly mandates a test case: 「`tests/resolve-batch-query.bats` に「空白を含む status 値でパースエラー」のケースを含める」 (a case exercising a whitespace-containing `status:` value). No such case exists among the 11 tests. The behavior itself is correct today (`--query "label:retro/* status:In progress"` exits 1), but it's regression-unprotected — the exit-1 is incidental fallthrough in the `*)` arm, not an explicit assertion, so a future refactor of that arm could silently break it with no test catching the regression. The Code Retrospective additionally states this Uncertainty item required no extra verification during implementation, which is inaccurate given the mandated test was never added.

Recommended fix: add a test asserting `status:In progress` (or similar whitespace-containing value) exits 1 with a `resolve-batch-query:` stderr message, per the Spec's own mandate.

File: scripts/resolve-batch-query.sh [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+done
+
+if [ "$HAS_LABEL" != true ] || [ -z "$LABEL_PATTERN" ]; then
+    echo "resolve-batch-query: query must include a non-empty label: clause" >&2
+    exit 1
+fi
+
+# (a) Fetch candidate Issues
+ISSUES_JSON=""
+if ! ISSUES_JSON=$(gh issue list --state open --json number,labels --limit "$LIMIT" 2>&1); then
```
Comment (saito) [inline scripts/resolve-batch-query.sh]: [Bug/Logic error] `ISSUES_JSON=$(gh issue list ... 2>&1)` merges `gh`'s stderr into the JSON payload even on a successful (exit 0) call. Downstream, line 121's `jq` runs inside a command substitution used in a `<<<` redirection, whose exit status does **not** trip `set -e`/`pipefail` — so any stderr pollution or jq parse failure degrades silently to "0 matching issues, exit 0" instead of an error. Verified by reproduction with a mock `gh` that writes one stderr line: output was empty, exit 0. Combined with Until mode step 4 ("empty output → converged"), a resolution failure becomes indistinguishable from genuine convergence, silently truncating the `--until` loop — the opposite of this script's documented fail-closed design.

Recommended fix: drop `2>&1` (capture stderr separately if needed for the exit-2 message), and explicitly check the jq step's exit status before feeding its output into the while loop.

File: skills/auto/SKILL.md [hunk 3]
```diff
@@ -1194,6 +1198,35 @@ After all Issues are processed, delete the batch checkpoint:
...
+5. Record the output as `ROUND_LIST`. Run:
+   ```
+   ${CLAUDE_PLUGIN_ROOT}/scripts/auto-checkpoint.sh write_batch "$BATCH_ID" "$ROUND_LIST" "$COMPLETED" "$FAILED"
+   ```
+   (this call also serves as List mode's own checkpoint initialization — do not call `write_batch` a second time). Add `ROUND_LIST` to `ALL_TARGETS`, then process each Issue in `ROUND_LIST` by applying `### List mode (--batch N1 N2 ...)` steps 1–7 unchanged. Whenever `update_batch ... complete` or `... fail` is called for an Issue number, add that number to `COMPLETED`/`FAILED` and to `PROCESSED`. An Issue skipped by the blocked-by gate (step 4 of List mode) is **not** added to `PROCESSED` — its blocker may clear within this same session, so it should be re-evaluated on the next round.
+6. If `CHECKIN_PER_ROUND` is `true` AND ARGUMENTS does **not** contain `--non-interactive`: use AskUserQuestion to confirm proceeding to the next round; any answer other than "continue" goes to step 7. If ARGUMENTS contains `--non-interactive`, output "Warning: --checkin-per-round ignored in non-interactive mode." and proceed without asking. In both cases (confirmed continue, or non-interactive skip), go back to step 2.
```
Comment (saito) [inline skills/auto/SKILL.md]: [Bug/Logic error] Until mode step 6 only defines an explicit "go back to step 2" edge for two cases: (a) `CHECKIN_PER_ROUND=true` + interactive, (b) `--non-interactive` present. The **default invocation** (`CHECKIN_PER_ROUND=false`, interactive, e.g. `/auto --batch --until "label:retro/*"` with no extra flags) matches neither clause, so a literal reading has no defined loop-back edge for the most common path — risking termination after a single round. Separately, the `--non-interactive` warning clause is not nested under `CHECKIN_PER_ROUND=true`, so it fires even when `--checkin-per-round` was never passed, contradicting `docs/workflow.md`'s own description (added in this same PR) that scopes the warning to the opt-in flag actually being used.

Recommended fix: restructure step 6 so the default `CHECKIN_PER_ROUND=false` path explicitly returns to step 2, and nest the `--non-interactive` warning under `CHECKIN_PER_ROUND=true`, e.g.: "If `CHECKIN_PER_ROUND` is `true`: (a) if not `--non-interactive`, AskUserQuestion...; (b) if `--non-interactive`, warn and proceed. If `false`, proceed without asking. In every case that did not go to step 7, go back to step 2."

File: skills/auto/SKILL.md [hunk 4]
```diff
@@ -1194,6 +1198,35 @@ After all Issues are processed, delete the batch checkpoint:
...
+   ```
+   - Exit 1 (parse error — the query itself is malformed): abort Until mode entirely (no `delete_batch` call needed if `write_batch` was never reached — see step 5).
+   - Exit 2 (`gh` failure): if `ROUND == 1`, abort Until mode; if `ROUND >= 2`, output a warning and treat as converged (go to step 7).
+4. If the output is empty: output "Until mode: query returned 0 issues at round $ROUND; converged." and go to step 7.
+5. Record the output as `ROUND_LIST`. Run:
+   ```
+   ${CLAUDE_PLUGIN_ROOT}/scripts/auto-checkpoint.sh write_batch "$BATCH_ID" "$ROUND_LIST" "$COMPLETED" "$FAILED"
+   ```
+   (this call also serves as List mode's own checkpoint initialization — do not call `write_batch` a second time). Add `ROUND_LIST` to `ALL_TARGETS`, then process each Issue in `ROUND_LIST` by applying `### List mode (--batch N1 N2 ...)` steps 1–7 unchanged. Whenever `update_batch ... complete` or `... fail` is called for an Issue number, add that number to `COMPLETED`/`FAILED` and to `PROCESSED`. An Issue skipped by the blocked-by gate (step 4 of List mode) is **not** added to `PROCESSED` — its blocker may clear within this same session, so it should be re-evaluated on the next round.
```
Comment (saito) [inline skills/auto/SKILL.md]: [Bug/Logic error] Step 5 delegates per-Issue processing to "`### List mode` steps 1–7 unchanged", but List mode's `AUTO_STOP_AT` is populated by a separate prerequisite block labeled "Load stop-at setting (List mode only)" that sits *outside* steps 1–7 (List mode's own Step 1 `--batch` branch explicitly skips Steps 2–6 where the general load lives). Since Until mode's step 5 only reuses steps 1–7, `AUTO_STOP_AT` is never initialized during Until mode, so List mode step 7's `AUTO_STOP_AT == "merge"` gate evaluates against an unset/defaulted value — a project configured with `auto-stop-at: merge` would have `wholework:verify` dispatched anyway during Until mode, diverging from List mode's own (correct) behavior. Note Resume mode does not share this gap — it references the whole List mode section unscoped.

Recommended fix: either widen Until mode step 5's reference to the full List mode section (matching Resume mode), or add a one-line stop-at load to Until mode step 1.

File: scripts/resolve-batch-query.sh [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+# treated fail-closed: that Issue is excluded (not a hard error) and a warning is printed to
+# stderr. The overall exit code stays 0 in that case.
+
+set -euo pipefail
+
+SCRIPT_DIR="${WHOLEWORK_SCRIPT_DIR:-$(cd "$(dirname "$0")" && pwd)}"
+
+QUERY=""
+EXCLUDE=""
+LIMIT=200
```
Comment (saito) [inline scripts/resolve-batch-query.sh:32]: [Logic] `LIMIT=200` bounds the `gh issue list --state open --json number,labels --limit "$LIMIT"` fetch, and label/status filtering happens client-side *after* this fetch (no server-side label filter is possible with glob patterns). On a repo with >200 open Issues, matching Issues beyond the 200 newest are invisible, and Until mode's convergence check ("empty output → converged") would report false convergence with no truncation warning anywhere. Unlike Count mode's `--limit 200` (which applies to an already server-side-filtered `--label triaged` set), here 200 applies to *all* open Issues, competing for slots with non-matching ones — a materially smaller effective window. This is directly relevant to the Issue's own stated primary use case: theme-driven consumption of *older* backlog Issues (`docs/product.md` § theme-driven backlog consumption), which are exactly the ones most likely to be pushed out of a newest-200 window on a busy repo. Currently latent (69 open Issues in this repo).

Recommended fix: emit a stderr warning when the fetched set size equals `$LIMIT` (indicating possible truncation), and/or document the cap as a known limitation in the Until mode section of `docs/workflow.md`.

**Review by saito — COMMENTED** [review]
Note: posted as COMMENT instead of REQUEST_CHANGES — self-review (own pull request) is not allowed to request changes.

## Acceptance Criteria Verification Results

| Condition | Result | Notes |
|-----------|--------|-------|
| rubric: `--batch --until` の条件駆動ループ (クエリ→List mode処理→再クエリ→空になるまで) が実装されている | PASS | `skills/auto/SKILL.md` の `### Until mode` セクション (Step 1-7) で確認 |
| rubric: 既定で有効な安全弁 (`--max-rounds`) が実装されている | PASS | `--max-rounds` 既定 3、opt-out 不可を確認 |
| `file_contains "skills/auto/SKILL.md" "--max-rounds"` | PASS | — |
| rubric: `auto-checkpoint.sh`/L3 auto-retrospective を List mode から再利用し重複実装がない | PASS | Step 5/7 で List mode 手順の verbatim 再利用を明記 (ただし Step 5 の再利用範囲「steps 1-7」に List mode 側の stop-at ロード漏れあり — Code Review 参照) |
| `command "bats tests/resolve-batch-query.bats"` | PASS | CI ジョブ `Run bats tests` の SUCCESS 経由で確認 (safe mode の CI reference fallback) |
| `file_contains "docs/workflow.md" "--until"` | PASS | — |
| (Post-merge, opportunistic) 実運用での安全弁動作観察 | POST-MERGE | 次回 `--batch --until` 実運用時に `/verify` が観察 |

## CI Status

| Job name | Status | Notes |
|---------|--------|-------|
| DCO | SUCCESS | — |
| Run bats tests (×2) | SUCCESS | — |
| Validate skill syntax (×2) | SUCCESS | — |
| Forbidden Expressions check (×2) | SUCCESS | — |
| Language Convention check (×2) | SUCCESS | — |
| macOS shell compatibility (×2) | SUCCESS | — |

All 11 checks SUCCESS.

## Base Branch Conflict Pre-check

`git merge-tree` (deprecated 3-arg form) flagged `docs/structure.md` / `docs/ja/structure.md` as "changed in both" against `origin/main` (both `[SSoT]`, matching `docs/*.md`). Independently verified via `git merge-tree --write-tree origin/main HEAD`: **no real conflict** — the merged tree cleanly contains both origin/main's `guard-prefix.sh` description update and this PR's `resolve-batch-query.sh` addition (non-overlapping lines, ~19 lines apart in the same "Key modules" list). All 3 review agents independently confirmed no genuine issue in these two files. No action needed.

## Code Review

### General Comments (issues where path/line cannot be identified)

**[Steering Documents Policy] docs/product.md — status: clause graceful degradation undocumented**
`docs/product.md` documents the GitHub Projects board as an optional dependency, falling back to label-only operation when absent. The `--until` query's `label:`-only path honors this correctly (no GraphQL call). However, the `status:` clause does not degrade gracefully without a board — every Issue hits fail-closed exclusion, and Until mode reports "converged" rather than surfacing "status filtering unsupported". This gap is undocumented. Severity: CONSIDER

Recommended fix: add a note to the `status:<value>` description in `docs/workflow.md` / `docs/ja/workflow.md` that it requires a GitHub Projects board and unresolvable Status values are excluded fail-closed.

**[Prose-Literal Inconsistency] skills/auto/SKILL.md — ROUND_LIST separator description**
Until mode step 5 passes `ROUND_LIST` (newline-separated, per `resolve-batch-query.sh`'s documented output format) to `write_batch`, but the format literal for the same argument elsewhere in the file (List mode's own checkpoint description) states "space-separated ... quoting required when the list contains spaces". No behavioral impact (`_numbers_to_json_array`'s `for n in $list` splits on default IFS, handling both separators identically), but a reader could reasonably misinterpret the contract. Severity: CONSIDER

Recommended fix: generalize the format literal to "whitespace-separated (spaces or newlines)", or add a one-line clarification at Until mode step 5.

### Issue Count Summary

- Posted as line comments: 7 (MUST: 1, SHOULD: 4, CONSIDER: 2)
- General Comments (in body): 2 (CONSIDER: 2)

### Issues Rejected by Verification (false positive filtering results)

Step 10.3 ran a 2-stage adversarial verification pass (Opus) over all `review-bug` findings before inclusion above. 2 of 9 candidate findings were rejected:

- Rejected: `scripts/resolve-batch-query.sh`'s Status-resolution pipeline (`... | head -1 | tr -d '"'`) could SIGPIPE the GraphQL producer under `pipefail`, causing a false fail-closed exclusion → Reason: empirically measured 0/300 SIGPIPE occurrences at realistic payload sizes (query is bounded to ≤10 short lines); the claimed failure mode requires ~1000+ lines of producer output, which this query structurally cannot produce.
- Rejected: Until mode step 6's `--checkin-per-round` confirmation prompt fires even on the last permitted round (before `--max-rounds` stops the loop anyway) → Reason: confirmed the mechanic is real but has zero functional impact — both "continue" and "stop" answers land on step 7 with identical state; cosmetic only.

---
MUST issues found. After fixing in Step 11 and posting response summary in Step 13, run `/merge 1333`.

File: skills/auto/SKILL.md (hunk 4 above)
Comment (saito) [inline skills/auto/SKILL.md]: [Bug/Logic error] Step 5 says "Add `ROUND_LIST` to `ALL_TARGETS`" (plain append), while step 7 describes `ALL_TARGETS` as "the union of every round's targets" (implies dedup). Step 5 also deliberately keeps blocked-by-gate-skipped Issues out of `PROCESSED` so they legitimately reappear in a later round's `ROUND_LIST` — and `resolve-batch-query.sh`'s `--exclude` only filters against `PROCESSED`, so nothing else deduplicates. A literal "append" implementation would duplicate that Issue number in `BATCH_LIST`, causing the Batch Completion Report to double-count it (duplicate report lines, inflated phase/verify counts, doubled verify-type totals). Low severity: this is a prose-instruction ambiguity affecting only the aggregate report, not the actual work performed — an executor reading step 7's "union" first would likely dedupe correctly anyway.

Recommended fix: reword step 5 to "Add each number in `ROUND_LIST` to `ALL_TARGETS` if not already present (set union)".

File: scripts/resolve-batch-query.sh [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+while [ $# -gt 0 ]; do
+    case "$1" in
+        --query)
+            QUERY="${2:-}"
+            shift 2
```
Comment (saito) [inline scripts/resolve-batch-query.sh:38]: [Bug/Error handling] Each option branch (`--query`/`--exclude`/`--limit`) does `shift 2` unconditionally. When the flag is the last CLI argument with no value, `shift 2` fails under `set -e` and the script aborts with exit 1 and **zero** stderr output — verified by direct reproduction (`bash scripts/resolve-batch-query.sh --query` → exit 1, 0 bytes stdout+stderr). This contradicts the script's own documented contract (header: parse errors print `resolve-batch-query: <reason>` to stderr). Low severity: the only production caller (`skills/auto/SKILL.md` step 3) always passes both `--query "$UNTIL_QUERY"` and `--exclude "$PROCESSED"` with quoted values, so this path is reachable only via manual/incorrect CLI invocation, not the actual `/auto` flow.

Recommended fix: guard each branch, e.g. `[ $# -ge 2 ] || { echo "resolve-batch-query: --query requires a value" >&2; exit 1; }` before shifting (same pattern as `scripts/gh-graphql.sh`).

**Comment by saito** [comment]
## Review Response Summary

### Claude Review Response

| Aspect | Filename | Line | Issue | Response | Fix/Skip reason |
|--------|---------|------|-------|----------|----------------|
| Spec deviation | tests/resolve-batch-query.bats | 158 | Spec-mandated whitespace-status parse-error test case missing | Resolved | Added `status value containing whitespace: unsupported, exit 1 as documented` test (12/12 pass) |
| Bug/Logic error | scripts/resolve-batch-query.sh | 97 | `gh` stderr merged into JSON payload + jq exit status discarded → resolution failures indistinguishable from genuine convergence | Resolved | Dropped `2>&1`, added explicit jq exit-status check before feeding the label filter loop |
| Bug/Logic error | skills/auto/SKILL.md | 1223 | Until mode step 6 had no explicit loop-back edge for the default (non-checkin, interactive) path; `--non-interactive` warning fired unconditionally | Resolved | Restructured step 6 so every non-step-7 path explicitly returns to step 2; nested the warning under `CHECKIN_PER_ROUND=true` |
| Bug/Logic error | skills/auto/SKILL.md | 1222 | Until mode's "steps 1-7 unchanged" delegation excluded List mode's separate stop-at load block, so `AUTO_STOP_AT` was never initialized | Resolved | Load `AUTO_STOP_AT` once in Until mode step 1 |
| Logic | scripts/resolve-batch-query.sh | 32 | `--limit 200` client-side filtering could silently truncate results on repos with >200 open Issues, with no warning | Resolved | Added a stderr warning when the fetched candidate set size equals `--limit` |
| Bug/Logic error | skills/auto/SKILL.md | 1222 | `ALL_TARGETS` accumulation described as plain "add" vs. step 7's "union", risking duplicate Batch Completion Report entries | Resolved | Reworded step 5 to an explicit set union |
| Bug/Error handling | scripts/resolve-batch-query.sh | 38 | `shift 2` on `--query`/`--exclude`/`--limit` aborts silently (no stderr) when the flag is the last CLI argument | Skipped | CONSIDER severity; unreachable via the only production caller (`skills/auto/SKILL.md`, which always passes both args quoted); deferred |
| Steering Documents Policy | docs/product.md | — | `status:` query clause doesn't degrade gracefully without a GitHub Projects board (undocumented) | Skipped | CONSIDER severity; documentation-only gap, deferred |
| Prose-Literal Inconsistency | skills/auto/SKILL.md | ~1151 vs 1222 | `ROUND_LIST` described as space-separated in one place, newline-separated in Until mode's actual usage | Skipped | CONSIDER severity; no behavioral impact (both separators handled identically by the consumer), deferred |

### Lightweight Re-check Results

- `bats tests/auto-batch.bats tests/resolve-batch-query.bats`: 32/32 PASS
- `bats tests/auto-*.bats`: 55/55 PASS
- `python3 scripts/validate-skill-syntax.py skills/auto/SKILL.md`: 0 errors, 0 warnings
- `bash scripts/check-forbidden-expressions.sh`: clean
- `bash scripts/check-allowed-tools.sh`: clean
- New issues: none

### Verification Methodology Note

Step 10 ran the static Task fan-out (review-spec + review-bug×2) rather than the Workflow tool path, despite `capabilities.workflow: true` being set — this session has no re-invocation guarantee (headless, non-interactive), and `skills/review/workflow-guidance.md` requires falling back to the static path in that case. All `review-bug` findings passed through a 2-stage Opus adversarial verification pass (7 confirmed, 2 rejected as false positives — see the Step 11 review body's "Issues Rejected by Verification" section) before inclusion above.
